### ISSUE TITLE
logit output of RandomForest

### ISSUE THREAD

**Issue by Howard-ll** [tensorflow/decision-forests#33]
Differently from GBT, it seems that RandomForest does not have logit output

The logits are available in v0.1.7, but the signature is different from sklearn:
# Trains a Gradient Boosted Trees that returns logits (assuming the dataset is a binary classification)
model = tfdf.keras.GradientBoostedTreesModel(apply_link_function=False)

**Comment by achoum** [comment]
Unlike Gradient Boosted Trees learning algorithm, the [Random Forest](https://www.stat.berkeley.edu/~breiman/randomforest2001.pdf) algorithm works with a "voting mechanism". In the case of classification, each tree casts a vote for one class (or multiple classes depending on the hyper-parameters). Therefore, the algorithm does not rely on any link functions / logits. This is why the argument `apply_link_function` does not exist for the Random Forest model.

I rarely saw logits being used with Random Forests. Out of curiosity, do you mind detailing your setup :) ?

If a logit is what you need (i.e. the inverse of the logistic function), you can always compute it from the probabilities (be careful with numerical precision and proba=0 case).

**Comment by Howard-ll** [comment]
Thanks for the answer. This request is about the number of unique output values. 

I am trying to replace a library with the tfdf. As you can see below screen capture, predict, predict_proba & predict_log_proba give me different output values. I am talking about the number of unique output values.
For my project, I need predict_proba or predict_log_proba. I understand that tfdf predict is similar to predict_proba. This is good. However, if I could get more number of unique output values, that would be really great. As you can see picture 2 & 3 below, predict_proba of sklearn has bigger number of unique output values while tfdf has just many 0s. If this feature can be supported, that will be just great to me because I do need it for my tasks. 

** 1) screen capture **
![Capture](https://user-images.githubusercontent.com/85953851/124657618-675f3a00-deab-11eb-963b-6c484fd04c13.JPG)

** 2) Library S output range **
![image](https://user-images.githubusercontent.com/85953851/124663531-e4da7880-deb2-11eb-9b0f-012063ea53b3.png)

** 3) tfdf output range **
![image](https://user-images.githubusercontent.com/85953851/124664191-c032d080-deb3-11eb-9f12-a35cec26d7c5.png)

**Comment by Howard-ll** [comment]
Need to try more data-sets but Random Forest has a good enough number of unique output values on my current test data-sets. Thanks!

**Comment by achoum** [comment]
Some quick remarks.

- The number of unique values would be the same for the logits and for the probabilities.
- Some hyper-parameters and the diversity of the dataset examples will impact the number of unique prediction values. For example, increasing the number and depth of the trees (I see that you train RF tree to only max depth 4) will help.
